datatypes/src/primitives/time_interval.rs

```diff
@@ -181,16 +218,16 @@ impl TimeInterval {
             }
         );
         Ok(Self {
-            start: i64::min(self.start, other.start),
-            end: i64::max(self.end, other.end),
+            start: i64::min(self.start.inner(), other.start.inner()).into(),
+            end: i64::max(self.end.inner(), other.end.inner()).into(),
         })
     }
 
-    pub fn start(&self) -> i64 {
+    pub fn start(&self) -> TimeInstance {
         self.start
     }
 
-    pub fn end(&self) -> i64 {
+    pub fn end(&self) -> TimeInstance {
```

Comment:
I don't like the idea. Work with timeinstance where possible and into() as_int() if you need thebasic type.
Where is this needed ?

Reply:
I guess we have to see this. It is important to keep TimeInstance an `i64` basically with no additions on the type to allow pointer casts. The suggestion is more about code ergonomics because I saw lots of `.into()`s everywhere.

Reply:
most into calls are gone. You can always use .inner() to make clear what happens.